get_databricks_cluster_eventlog.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+# Default directory to save results to
+results_dir="$PWD/databricks_cluster_eventlogs"
+mkdir -p $results_dir
+
+function print_usage() {
+  cat <<-EOD
+	Usage: $0 -c <clusterId> [-r <results directory>]
+	Options:
+	  -c <cluster ID>  Databricks cluster ID
+	  -r <results directory>  directory in which to save cluster eventlog
+	  -h                      show this helpful message and exit
+	EOD
+}
+
+while getopts c:r:h: flag
+do
+    case "${flag}" in
+        c) clusterid="$OPTARG";;
+        r) results_dir="$OPTARG";;
+        h) print_usage && exit;;
+        *) >&2 print_usage && exit 1;;
+    esac
+done
+
+
+offset=0
+function eventlog_cli_call {
+    databricks clusters events --cluster-id $clusterid --output JSON --order ASC --offset $offset
+}
+
+
+# Make initial databricks call
+call=$(eventlog_cli_call)
+events=$(jq '.events' <<< $call)
+total_count=$(jq '.total_count' <<< $call)
+
+# Repeat databricks call until there's no "next_page"
+while $(jq 'has("next_page")' <<< $call)
+do
+    offset=$(jq '.next_page.offset' <<< $call)
+    call=$(eventlog_cli_call)
+    new_events=$(jq '.events' <<<$call)
+
+    # Append the new_events onto the events list
+    events=$(jq --argjson x "$events" --argjson y "$new_events" -n '$x + $y')
```

Comment:
Depending on how many events we'll see JSON lines might be a better output format.

Reply:
I'm not certain what you mean by this. Is there a concept of a json variable in bash? Or do you mean to save the results  to a temporary json file on each iteration? Or something else?

Reply:
I mean the format that we expect for event logs in the application: https://jsonlines.org/, e.g.
```
{"Event":"DBCEventLoggingListenerMetadata","Spark Version":"...
{"Event":"SparkListenerResourceProfileAdded","Resource Profi...
{"Event":"SparkListenerBlockManagerAdded","Block Manager ID"...
{"Event":"SparkListenerEnvironmentUpdate","JVM Information":...
{"Event":"SparkListenerApplicationStart","App Name":"Databri...
```
It would make concatenating pages easier too using `>>` to append to the output file.

Reply:
I see what you mean. As it stands now I maintained the format of the log as it comes from databricks. Example here. I could have it print each cli call as a JSON line as you say, but then we'd have make up for it with some extra processing on our end, so I'm not sure it's worth it unless we run into issues.

```
{
  "events": [
    {
      "cluster_id": "0805-161921-aqhwkli",
      "timestamp": 1659716361670,
      "type": "CREATING",
      "details": {
        "cluster_size": {
          "num_workers": 4
        },
        "user": "sean.gorsky@synccomputing.com",
        "job_run_name": "job-712309320163671-run-10841-cluster-eventlog-test3"
      }
    },
    {
      "cluster_id": "0805-161921-aqhwkli",
      "timestamp": 1659716454904,
      "type": "INIT_SCRIPTS_STARTED",
      "details": {
        "init_scripts": {
          "cluster": [
            {
              "dbfs": {
                "destination": "dbfs:/databricks/scripts/tpcds-install.sh"
              }
            }
          ]
        }
      }
    },
    {
      "cluster_id": "0805-161921-aqhwkli",
      "timestamp": 1659716489735,
      "type": "INIT_SCRIPTS_FINISHED",
      "details": {
        "init_scripts": {
          "reported_for_node": "0805-161921-aqhwkli_10_152_228_47",
          "cluster": [
            {
              "dbfs": {
                "destination": "dbfs:/databricks/scripts/tpcds-install.sh"
              },
              "status": "SUCCEEDED",
              "execution_duration_seconds": 18
            }
          ]
        }
      }
    },
    {
      "cluster_id": "0805-161921-aqhwkli",
      "timestamp": 1659716511020,
      "type": "RUNNING",
      "details": {
        "current_num_workers": 4,
        "target_num_workers": 4
      }
    },
    {
      "cluster_id": "0805-161921-aqhwkli",
      "timestamp": 1659717813212,
      "type": "TERMINATING",
      "details": {
        "reason": {
          "code": "JOB_FINISHED",
          "type": "SUCCESS"
        }
      }
    }
  ],
  "total_count": 5
}
```

Reply:
I think if we were to use JSON Lines we'd want each event to be on its own line instead of each CLI response object. Though I understand keeping it as is if that's what the app already expects.

Reply:
Ok, I prefer to keep it as is for now, then.